Lib/unittest/loader.py

```diff
@@ -466,6 +475,10 @@ def _find_test_path(self, full_path, pattern, namespace=False):
             load_tests = None
             tests = None
             name = self._get_name_from_path(full_path)
+
+            if not name.isidentifier():
+                return None, False
```

Comment:
Note, AFAICT, Robert Collins just wanted to make this fix available in the unittest2 package, not backport it to the Pyhton2 branch.  Since he hasn't replied, though, I'll go ahead and remove the python2-isms.